tests/pytests/test_optking.py

```diff
@@ -271,3 +272,108 @@ def test_dimers_ne2_long(inp):
+    ('EXT_FORCE_DISTANCE', "`1 2 SIN(X)`"),
+    ('EXT_FORCE_OOFP', "`1 2 3 4 SIN(X)`"),
+    ('FIX_VAL_NEAR_PI', 0.01),
+    ('FLEXIBLE_G_CONVERGENCE', True),
+    ('FRAG_MODE', "`MULTI`"),
+    # ('FRAG_REF_ATOMS', ),
+    ('FREEZE_ALL_DIHEDRALS', True),
+    ('FREEZE_INTRAFRAG', True),
+    ('FROZEN_BEND', "1 2 3"),
+    ('FROZEN_CARTESIAN', "1 XY"),
+    ('FROZEN_DIHEDRAL', "1 2 3 4"),
+    ('FROZEN_DISTANCE', "1 2"),
+    ('FROZEN_OOFP', "1 2 3 4"),
+    ('FULL_HESS_EVERY', 6),
+    ('GEOM_MAXITER', 7),
+    ('G_CONVERGENCE', "CFOUR"),
+    ('HESSIAN_FILE', pathlib.Path("random_file.hess")),
+    ('HESS_UPDATE', "BOFILL"),
+    ('HESS_UPDATE_DEN_TOL', 0.01),
+    ('HESS_UPDATE_DQ_TOL', 0.01),
+    ('HESS_UPDATE_LIMIT', False),
+    ('HESS_UPDATE_LIMIT_MAX', 0.01),
+    ('HESS_UPDATE_LIMIT_SCALE', 0.01),
+    ('HESS_UPDATE_USE_LAST', 12),
+    ('H_BOND_CONNECT', True),
+    ('H_GUESS_EVERY', False),
+    ('INCLUDE_OOFP', True),
+    ('INTERFRAG_COLLINEAR_TOL', 0.01),
+    # ('INTERFRAG_COORDS', ), tested elsewhere
+    ('INTERFRAG_DIST_INV', True),
+    ('INTERFRAG_HESS', "FISCHER_LIKE"),
+    ('INTERFRAG_MODE', "PRINCIPAL_AXES"),
+    ('INTERFRAG_STEP_LIMIT', 0.01),
+    ('INTERFRAG_STEP_LIMIT_MAX', 0.01),
+    ('INTERFRAG_STEP_LIMIT_MIN', 0.01),
+    ('INTRAFRAG_HESS', "LINDH"),
+    ('INTRAFRAG_STEP_LIMIT', 0.01),
+    ('INTRAFRAG_STEP_LIMIT_MAX', 0.01),
+    ('INTRAFRAG_STEP_LIMIT_MIN', 0.01),
+    ('IRC_CONVERGENCE', -0.9),
+    ('IRC_DIRECTION', "BACKWARD"),
+    ('IRC_MODE', "CONFIRM"),
+    ('IRC_POINTS', 30),
+    ('IRC_STEP_SIZE', 0.01),
+    ('LINEAR_ALGEBRA_TOL', 0.01),
+    ('LINEAR_BEND_THRESHOLD', 0.01),
+    ('LINESEARCH', True),
+    ('LINESEARCH_STEP', 0.01),
+    ('MAX_DISP_G_CONVERGENCE', 0.01),
+    ('MAX_ENERGY_G_CONVERGENCE', 0.01),
+    ('MAX_FORCE_G_CONVERGENCE', 0.01),
+    ('OPT_COORDINATES', "CARTESIAN"),
+    ('OPT_TYPE', "TS"),
+    ('PRINT', 4),
+    ('RANGED_BEND', "1 2 3 120 130"),
+    ('RANGED_CARTESIAN', "1 XY 1.5 1.7"),
+    ('RANGED_DIHEDRAL', "1 2 3 4 120 130"),
+    ('RANGED_DISTANCE', "1 2 1.2 1.4"),
+    ('RANGED_OOFP', "1 2 3 4 60.0 65.0"),
+    ('REDUNDANT_EVAL_TOL', 0.01),
+    ('RFO_FOLLOW_ROOT', True),
+    ('RFO_NORMALIZATION_MAX', 0.01),
+    ('RFO_ROOT', 4),
+    ('RMS_DISP_G_CONVERGENCE', 0.01),
+    ('RMS_FORCE_G_CONVERGENCE', 0.01),
+    ('RSRFO_ALPHA_MAX', 0.01),
+    ('SD_HESSIAN', 0.01),
+    ('SIMPLE_STEP_SCALING', True),
+    ('SMALL_BEND_FIX_THRESHOLD', 0.01),
+    ('STEEPEST_DESCENT_TYPE', "BARZILAI_BORWEIN"),
+    ('STEP_TYPE', "NR"),
+    ('TEST_B', True),
+    ('TEST_DERIVATIVE_B', True),
+    ('UNFREEZE_DIHEDRALS', "1 2 3 4"),
+    ('V3D_TORS_ANGLE_LIM', 0.01),
+    ('V3D_TORS_COS_TOL', 0.01),
+    ('WRITE_TRAJECTORY', True)]
+)
+def test_all_keywords(option, test_val):
+
+    import optking
+    params = optking.op.OptParams(**{option: test_val})
+    processed = params.to_dict(by_alias=True)
+    assert processed.get(option) == test_val
```

Comment:
nice and reassuring. I imagine optking itself checks that bad values get sensible errors?